scipy/interpolate/tests/test_rgi.py

```diff
@@ -688,6 +689,34 @@ def test_bad_solver(self):
                 (x, y), data, method='slinear',  solver_args={'woof': 42}
             )
 
+    def test_concurrency(self):
+        barrier = threading.Barrier(10)
+        points, values = self._get_sample_4d()
+        sample = np.array([[0.1 , 0.1 , 1.  , 0.9 ],
+                           [0.2 , 0.1 , 0.45, 0.8 ],
+                           [0.5 , 0.5 , 0.5 , 0.5 ]])
+        interp = RegularGridInterpolator(points, values, method="slinear")
```

Comment:
Is it that all spline methods share the implementation, thus `method="slinear"` is enough to test them all? Or is it just that the real test below tests `__call__` and it basically does not matter how exactly the spline is constructed. Either way, please add a comment.

Reply:
This tests that the `method` attribute defined in the spline does not mutate under a concurrent scenario, which was happening before this PR